### PR TITLE
handle package split post 24.2 for airgap playbooks

### PR DESCRIPTION
**Pull request by jackietung-redpanda**

### CODE REVIEW

**Comment by jackietung-redpanda** [comment]
@gene-redpanda  I have made a stab at handling package split (as of 24.2), for debs so far.

If you are comfortable with this pattern, I will flesh out rpm as well.  Then we are done (for now)

**Comment by jackietung-redpanda** [comment]
Note: I did try to write a custom filter (e.g. `get_deb_url`) in python code to avoid more craziness with defining even more URL constants.  However I could not figure out a way to do it in our scenario (i.e. "embed the custom filter function within `redpanda-ansible-collection` AND also calling it from within `redpanda-ansible-collection`, but the invoking playbook living outside (in `deployment-automation`).

I relented and just added some extra variables.

File: roles/redpanda_broker/tasks/main.yml [hunk 1]
```diff
@@ -22,37 +22,37 @@
...
     - ansible_os_family == 'Debian'
-    - enable_airgap is false
+    - not (enable_airgap | bool)
 
 - name: Install Redpanda RPM
   ansible.builtin.include_tasks: install-rp-rpm.yml
   when:
     - ansible_os_family == 'RedHat'
-    - enable_airgap is false
+    - not (enable_airgap | bool)
```
Comment (gene-redpanda) [inline roles/redpanda_broker/tasks/main.yml:37]: why the difference between here and L49?

**Comment by gene-redpanda** [comment]
This looks pretty good to me. Have you done manual testing with it to validate outcomes?

**Comment by gene-redpanda** [comment]
> Note: I did try to write a custom filter (e.g. `get_deb_url`) in python code to avoid more craziness with defining even more URL constants. However I could not figure out a way to do it in our scenario (i.e. "embed the custom filter function within `redpanda-ansible-collection` AND also calling it from within `redpanda-ansible-collection`, but the invoking playbook living outside (in `deployment-automation`).
> 
> I relented and just added some extra variables.

Yeah intent is to support the prior pattern re: deployment-automation which does mean things get a little tricky sometimes

**Comment by jackietung-redpanda** [comment]
Yes I tested manually by running the deb bundling and the airgap provisioning (for debs).  Works fine.  I will port the approach to rpm as well and publish the PR.

**Comment by jackietung-redpanda** [comment]
Manual testing setup:
* AWS
* For both RPM (Fedora OS) and DEB (Ubuntu OS):
   * pre 24.2, regular install
   * pre 24.2, airgap install
   * 24.2, regular install
   * 24.2, airgap install

**Review by gene-redpanda — APPROVED** [review]

**Comment by gene-redpanda** [comment]
@jackietung-redpanda add a bump to the Galaxy version then feel free to merge it. I will cut the release for you (haven't had time to automate it yet 😅)

**Pull request merged**